Tests/FluentAssertions.Specs/Specialized/TaskCompletionSourceAssertionSpecs.cs

```diff
@@ -26,7 +120,7 @@ public async Task When_TCS_completes_in_time_it_should_succeed()
     }
 
     [Fact]
-    public async Task When_TCS_completes_in_time_and_result_is_expected_it_should_succeed()
+    public async Task When_generic_TCS_completes_in_time_and_result_is_expected_it_should_succeed()
```

Comment:
🔧  Instead of prefixing every test case with _When generic TCS_ or _When TCS_, put both sets of tests in a nested class. 
🔧 I would also just replace _TCS_ with _it_ or something shorter and functional.